mbedtls/src/pk/mod.rs

```diff
@@ -1259,23 +1259,42 @@ iy6KC991zzvaWY/Ys+q/84Afqa+0qJKQnPuy/7F5GkVdQA/lfbhi
         ];
 
         for digest in &digests {
+            let digest = *digest;
```

Comment:
Making changes to the test is not appropriate as this implies a public API change, which is backwards-incompatible. Instead, change `Pk::sign` to deal with the new behavior. The release notes imply this is a RSA-specific change, so please also add some ECDSA tests to make sure you're not making any changes to the behavior there.

Reply:
Also please add a test with `hash=&[]` as the release notes indicate special before if zero.

Reply:
What kind of changes are you expecting in `Pk::sign`? AFAIU, the change was a bugfix for a behaviour that was already incorrect. This change was backported in `2.26` requiring the argument `hash_len` to be at least the length of message digest. In `2.28`, the length needs to be equal.
Passing a lesser length is surely a bug but we could probably add a patch to `vendor` to accept `hash_len` which is at least the length of message digest instead of equal and bump the patch version of `mbedtls` crate?

References:
[1] https://github.com/ARMmbed/mbedtls/pull/4561
[2] https://github.com/ARMmbed/mbedtls/pull/4707

Reply:
When I wrote my comment I was thinking we could make `hash` shorter in sign/verify. But actually, silently signing less than the full hash input by the user is also bad. So I think we need to make the incompatible change for security reasons. I've simplified your changes though (force-pushed onto your branch), please review.

Reply:
Thanks, those changes LGTM.